client.py

```diff
@@ -87,12 +89,11 @@ def main():
                 if ('cpu_only' in config_cluster and config_cluster['cpu_only']
                     and s["gpus"] != 0):
                     continue
-                if (i >= config_cluster["limit_per_submit"]
-                    or i + glideins_running >= config_cluster["max_total_jobs"]):
+                if (i >= limit):
                     logger.info('reached limit')
                     break
-                if s["count"] > config_cluster["limit_per_submit"]: 
-                    s["count"] = config_cluster["limit_per_submit"]
+                if s["count"] > limit: 
+                    s["count"] = limit
```

Comment:
Wouldnt we have to pop the state list in that case? It seems better to just
loop through the available states and break in case we reached the limit.

On 17 March 2016 at 10:34, David Schultz <notifications@github.com> wrote:

> In client.py
> <https://github.com/dsschult/pyglidein/pull/11#discussion_r56534809>:
>
> >                      logger.info('reached limit')
> >                      break
> > -                if s["count"] > config_cluster["limit_per_submit"]:
> > -                    s["count"] = config_cluster["limit_per_submit"]
> > +                if s["count"] > limit:
> > +                    s["count"] = limit
>
> We could also think of just decrementing limit instead of incrementing i.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dsschult/pyglidein/pull/11/files/73d88b6c5c951af159db6269b845bb4707f00dfe..8d91d48dc8826bee61f46fcf01f2cfab794e8637#r56534809>
>



-- 
Benedikt Riedel
Postdoctoral Fellow
University of Alberta
Department of Physics